compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -167,10 +167,10 @@ fn compare_method_predicate_entailment<'tcx>(
     //
     // FIXME(@lcnr): remove that after removing `cause.body_id` from
     // obligations.
-    let impl_m_hir_id = tcx.hir().local_def_id_to_hir_id(impl_m.def_id.expect_local());
+    let impl_m_def_id = impl_m.def_id.expect_local();
     let cause = ObligationCause::new(
         impl_m_span,
-        impl_m_hir_id,
+        impl_m.def_id.expect_local(),
         ObligationCauseCode::CompareImplItemObligation {
             impl_item_def_id: impl_m.def_id.expect_local(),
```

Comment:
```suggestion
            impl_item_def_id: impl_m_def_id,
```

Reply:
There's some more occurrences further down as well.

Reply:
this is still unaddressed

Reply:
this is still unaddressed